contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--3.3.0--3.4.0.sql

```diff
@@ -298,6 +298,55 @@ ALTER FUNCTION sys.power(IN arg1 SMALLINT, IN arg2 NUMERIC) STRICT;
 
 ALTER FUNCTION sys.power(IN arg1 TINYINT, IN arg2 NUMERIC) STRICT;
 
+-- Update data-type of information_schema_tsql.TABLE_TYPE to sys.varchar if it's data-type is pg_catalog.varchar
+DO
+$$
+BEGIN  
+
+    IF EXISTS(
+        SELECT 1
+        FROM information_schema.columns
+        WHERE table_schema='information_schema_tsql'
+            AND table_name='tables'
+            AND column_name='TABLE_TYPE'
+            AND udt_schema='pg_catalog'
+            AND udt_name='varchar'
+    ) THEN
+        ALTER VIEW information_schema_tsql.tables RENAME TO tables_deprecated_in_3_4_0;
+
+        CREATE OR REPLACE VIEW information_schema_tsql.tables AS
+            SELECT CAST(nc.dbname AS sys.nvarchar(128)) AS "TABLE_CATALOG",
+                CAST(ext.orig_name AS sys.nvarchar(128)) AS "TABLE_SCHEMA",
+                CAST(
+                    CASE WHEN c.reloptions[1] LIKE 'bbf_original_rel_name%' THEN substring(c.reloptions[1], 23)
+                        ELSE c.relname END
+                    AS sys._ci_sysname) AS "TABLE_NAME",
+
+                CAST(
+                    CASE WHEN c.relkind IN ('r', 'p') THEN 'BASE TABLE'
+                        WHEN c.relkind = 'v' THEN 'VIEW'
+                        ELSE null END
+                    AS sys.varchar(10)) COLLATE sys.database_default AS "TABLE_TYPE"
+
+            FROM sys.pg_namespace_ext nc JOIN pg_class c ON (nc.oid = c.relnamespace)
+                LEFT OUTER JOIN sys.babelfish_namespace_ext ext on nc.nspname = ext.nspname
+
+            WHERE c.relkind IN ('r', 'v', 'p')
+                AND (NOT pg_is_other_temp_schema(nc.oid))
+                AND (pg_has_role(c.relowner, 'USAGE')
+                    OR has_table_privilege(c.oid, 'SELECT, INSERT, UPDATE, DELETE, TRUNCATE, REFERENCES, TRIGGER')
+                    OR has_any_column_privilege(c.oid, 'SELECT, INSERT, UPDATE, REFERENCES') )
+                AND ext.dbid = cast(sys.db_id() as oid)
+                AND (NOT c.relname = 'sysdatabases');
+
+        GRANT SELECT ON information_schema_tsql.tables TO PUBLIC;
+
+        CALL sys.babelfish_drop_deprecated_object('view', 'information_schema_tsql', 'tables_deprecated_in_3_4_0');
+    END IF;
+END
+$$
+LANGUAGE plpgsql;
+
 -- Drops the temporary procedure used by the upgrade script.
 -- Please have this be one of the last statements executed in this upgrade script.
 DROP PROCEDURE sys.babelfish_drop_deprecated_object(varchar, varchar, varchar);
```

Comment:
Can we move these 3 lines at the end of this file?

Reply:
Done